scipy/signal/_upfirdn_apply.pyx

```diff
@@ -65,6 +65,56 @@ def _output_len(Py_ssize_t len_h,
     return need
 
 
+@cython.cdivision(True)  # faster modulo
+@cython.boundscheck(False)  # designed to stay within bounds
+@cython.wraparound(False)  # we don't use negative indexing
+def _rat(x, tol=None):
```

Comment:
> You'll need to limit the number of iterations in the loop. Calling _rat(10000000001/10000000000.0) never returns. While its true that arrays of that size would be enormous, its still not good to have this possibility here.

I think that's a bug in my Cythoning. My (much slower) Python equivalent gives:

[9999999173 9999999172]

But you're right it's probably still safest to limit the loop.

> What constitutes a reasonable filter size?

Good question. I'm not sure what a good metric would be. Maybe for now we should just use GCD, and advise people that really long filters could be used for long signals?